### ISSUE TITLE
bug: VNode properties are minified incorrectly

### ISSUE THREAD

**Issue by tricki** [ionic-team/stencil#3089]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/master/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/master/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.8.1

### Current Behavior

VNode properties get incorrectly minified and are therefore not accessible in production builds.

Example: I access `vnode.$attrs$` which gets minified to `t.l` but the minified VNode doesn't have an `l` property.

### Expected Behavior

Code is correctly minified.

### Steps to Reproduce

1. Write code to access any property of a VNode and log its value.
2. Create a minified (production) build.
3. The value is incorrect (the value of another property).

### Code Reproduction URL

https://github.com/tricki/stencil-vnode-minif-bug

### Additional Information

I originally encountered this bug in `@stencil/helmet` which doesn't work when minified because of this bug, see https://github.com/ionic-team/stencil-helmet/issues/15

**Comment by MarkChrisLevy** [comment]
Hi, this is how it works (obfuscation of VNode) from the very beginning of stencil. You cannot access VNode, but rather ChildNode, which you can access using utils argument of FunctionalComponent interface. More in the docs https://stenciljs.com/docs/functional-components.

**Comment by tricki** [comment]
@MarkChrisLevy Interesting, but then why would Helmet have implemented it like this in the first place? 

Based on your feedback I created a small workaround using a functional component (I couldn't find another way to access `FunctionalUtilities`):

```tsx
import { FunctionalUtilities, VNode, ChildNode } from '@stencil/core';

function GetChildNodes(_props, children: VNode[], utils: FunctionalUtilities): ChildNode[] {
  const newChildren = [];
  utils.forEach(children, child => newChildren.push(child));
  return newChildren;
}
```

**Usage:**

```tsx
import { ChildNode, h } from '@stencil/core';

const nodes = <GetChildNodes><div class="foo">bar</div></GetChildNodes>
nodes.forEach((node: ChildNode) => {
  // access node.vtag, node.vtext etc.
});
```

**Comment by MarkChrisLevy** [comment]
As far I can see Helmet is always using FunctionalUtilities not directly VNode.

**Comment by tricki** [comment]
You're right, [the buggy line](https://github.com/ionic-team/stencil-helmet/blob/master/src/stencil-helmet.ts#L9) does use `ChildNode` through [`utils.forEach()`](https://github.com/ionic-team/stencil-helmet/blob/master/src/stencil-helmet.ts#L23) but it still gets minified incorrectly, the details are in https://github.com/ionic-team/stencil-helmet/issues/15#issuecomment-922891293. Could the problem be that it's getting minified twice?
